RecoCTPPS/TotemRPLocal/plugins/TotemRPUVPatternFinder.cc

```diff
@@ -296,4 +297,58 @@ void TotemRPUVPatternFinder::produce(edm::Event& event, const edm::EventSetup& e
  
 //----------------------------------------------------------------------------------------------------
 
+void
+TotemRPUVPatternFinder::fillDescriptions( edm::ConfigurationDescriptions& descr )
+{
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>( "tagRecHit", edm::InputTag( "totemRPRecHitProducer" ) ); // input selection
+  desc.addUntracked<unsigned int>( "verbosity", 0 );
+  desc.add<unsigned int>( "maxHitsPerPlaneToSearch", 5 ); // if a plane has more hits than this parameter, it is considered as dirty
+  desc.add<unsigned int>( "minPlanesPerProjectionToSearch", 3 ); // minimal number of reasonable (= not empty and not dirty) planes per projeciton and per RP, to start the pattern search
+  desc.add<double>( "clusterSize_a", 0.02 /* rad */ ); // (full) cluster size in slope-intercept space
+  desc.add<double>( "clusterSize_b", 0.3 /* mm */ );
+
+  // minimal weight of (Hough) cluster to accept it as candidate
+  //   weight of cluster = sum of weights of contributing points
+  //   weight of point = sigma0 / sigma_of_point
+  //   most often: weight of point ~ 1, thus cluster weight is roughly number of contributing points
+  desc.add<double>( "threshold", 2.99 );
+
+  // minimal number of planes (in the recognised patterns) per projeciton and per RP, to tag the candidate as fittable
+  desc.add<unsigned int>( "minPlanesPerProjectionToFit", 3 );
+
+  // whether to allow combination of most significant U and V pattern, in case there several of them
+  // don't set it to True, unless you have reason
+  desc.add<bool>( "allowAmbiguousCombination", false );
+
+  // maximal angle (in any projection) to mark the candidate as fittable -> controls track parallelity with beam
+  // huge value -> no constraint
+  desc.add<double>( "max_a_toFit", 10.0 );
+
+  edm::ParameterSetDescription exceptions_validator;
+  exceptions_validator.add<unsigned int>( "rpId" ); // RP id according to CTPPSDetId
+  exceptions_validator.add<unsigned int>( "minPlanesPerProjectionToFit_U" );
+  exceptions_validator.add<unsigned int>( "minPlanesPerProjectionToFit_V" );
+  exceptions_validator.add<double>( "threshold_U" );
+  exceptions_validator.add<double>( "threshold_V" );
+
+  std::vector<edm::ParameterSet> exceptions_default;
+
+  // if a RP or projection needs adjustment of the above settings, you can use the following format:
```

Comment:
In b9f4ef0b26c66600112ec4d40dea97ffa5e7c830 the full CTPPS reconstruction example is amended to introduce (as comments) the usage of this set of exceptional settings to the U-V pattern finder.